### PR TITLE
[KOGITO-6087] - Create image for data-index-inmemory service

### PR DESCRIPTION
**Pull request by spolti**
Signed-off-by: spolti <fspolti@redhat.com>
See: https://issues.redhat.com/browse/KOGITO-6087

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a testcase that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by spolti** [comment]
tests will fail since there is not snapshot version of the inmemory artifact yet.

**Comment by spolti** [comment]
@vaibhavjainwiz do we need to make changes on operator to have this image working?

**Review by radtriste — COMMENTED** [review]
Could we call it `kogito-data-index-ephemeral` instead of `kogito-data-index-inmemory` like we did on jobs-service ?

File: tests/features/data-index/kogito-data-index-inmemory.feature [hunk 1]
```diff
@@ -0,0 +1,26 @@
+@quay.io/kiegroup/kogito-data-index-inmemory
+Feature: Kogito-data-index postgresql feature.
```
Comment (radtriste) [inline tests/features/data-index/kogito-data-index-inmemory.feature]: Please rename the Feature

File: tests/features/data-index/kogito-data-index-inmemory.feature [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      And the image should contain label io.k8s.description with value Runtime image for Kogito Data Index Service for in memory PostgreSQL persistence provider
+      And the image should contain label io.k8s.display-name with value Kogito Data Index Service - in memory PostgreSQL
+      And the image should contain label io.openshift.tags with value kogito,data-index,data-index-inmemory
+
+  Scenario: verify if the indexing service binaries are available on /home/kogito/bin
+    When container is started with command bash
+    Then run sh -c 'ls /home/kogito/bin/data-index-service-inmemory-runner.jar' in container and immediately check its output for /home/kogito/bin/data-index-service-inmemory-runner.jar
+
+    @wip
+  Scenario: verify if of container is correctly started with postgresql parameters
```
Comment (radtriste) [inline tests/features/data-index/kogito-data-index-inmemory.feature]: Please rename scenario

File: tests/features/data-index/kogito-data-index-postgresql.feature [hunk 3]
```diff
@@ -9,7 +9,7 @@ Feature: Kogito-data-index postgresql feature.
       And the image should contain label io.openshift.expose-services with value 8080:http
       And the image should contain label io.k8s.description with value Runtime image for Kogito Data Index Service for PostgreSQL persistence provider
       And the image should contain label io.k8s.display-name with value Kogito Data Index Service - PostgreSQL
-      And the image should contain label io.openshift.tags with value kogito,data-index,data-index-postgresql
```
Comment (radtriste) [inline tests/features/data-index/kogito-data-index-postgresql.feature]: why this change ?

**Review by radtriste — CHANGES_REQUESTED** [review]

File: tests/features/data-index/kogito-data-index-inmemory.feature (hunk 2 above)
Comment (sutaakar) [reply tests/features/data-index/kogito-data-index-inmemory.feature]: Also please remove wip tag

**Review by sutaakar — APPROVED** [review]

**Comment by sutaakar** [comment]
> do we need to make changes on operator to have this image working?

By default we use Data index Infinispan image by operator. We should switch to use this image.

**Review by cristianonicolai — APPROVED** [review]

**Comment by spolti** [comment]
> Could we call it `kogito-data-index-ephemeral` instead of `kogito-data-index-inmemory` like we did on jobs-service ?

I was thinking about use ephemeral as well, but I kept the same name from the artifact like we do with the other images.
but ephemeral makes more sense to me as well.

**Comment by spolti** [comment]
> By default we use Data index Infinispan image by operator. We should switch to use this image.

I'll be creating a follow up jira to address it.

**Comment by radtriste** [comment]
> > Could we call it `kogito-data-index-ephemeral` instead of `kogito-data-index-inmemory` like we did on jobs-service ?
> 
> I was thinking about use ephemeral as well, but I kept the same name from the artifact like we do with the other images. but ephemeral makes more sense to me as well.

we use already different for jobs-service https://github.com/kiegroup/kogito-images/blob/main/scripts/update-maven-artifacts.py#L37

**Comment by radtriste** [comment]
> > > Could we call it `kogito-data-index-ephemeral` instead of `kogito-data-index-inmemory` like we did on jobs-service ?
> > 
> > 
> > I was thinking about use ephemeral as well, but I kept the same name from the artifact like we do with the other images. but ephemeral makes more sense to me as well.
> 
> we use already different for jobs-service https://github.com/kiegroup/kogito-images/blob/main/scripts/update-maven-artifacts.py#L37

btw created https://github.com/kiegroup/kogito-images/pull/791 to switch from `jobs-service-common` to `jobs-service-inmemory` runner jar

File: tests/features/data-index/kogito-data-index-inmemory.feature (hunk 1 above)
Comment (spolti) [reply tests/features/data-index/kogito-data-index-inmemory.feature]: done

File: tests/features/data-index/kogito-data-index-inmemory.feature (hunk 2 above)
Comment (spolti) [reply tests/features/data-index/kogito-data-index-inmemory.feature]: done

File: tests/features/data-index/kogito-data-index-postgresql.feature (hunk 3 above)
Comment (spolti) [reply tests/features/data-index/kogito-data-index-postgresql.feature]: mistake :)

thanks

**Review by radtriste — APPROVED** [review]

File: tests/features/data-index/kogito-data-index-postgresql.feature [hunk 4]
```diff
@@ -23,6 +23,5 @@ Feature: Kogito-data-index postgresql feature.
       | QUARKUS_DATASOURCE_USERNAME  | kogito                                   |
       | QUARKUS_DATASOURCE_PASSWORD  | s3cr3t                                   |
     Then container log should contain -Djava.library.path=/home/kogito/lib -Dquarkus.http.host=0.0.0.0 -Dquarkus.http.port=8080 -jar /home/kogito/bin/data-index-service-postgresql-runner.jar
-     And container log should contain org.postgresql.util.PSQLException: The connection attempt failed
-     And container log should not contain Application failed to start
-
+    And container log should contain org.postgresql.util.PSQLException: The connection attempt failed
+    And container log should not contain Application failed to start
```
Comment (radtriste) [inline tests/features/data-index/kogito-data-index-postgresql.feature]: could you set the same for jobs-service-postgresql ?

**Review by sutaakar — APPROVED** [review]

File: tests/features/data-index/kogito-data-index-postgresql.feature (hunk 4 above)
Comment (spolti) [reply tests/features/data-index/kogito-data-index-postgresql.feature]: do you mean add these two checks there?

Comment (radtriste) [reply tests/features/data-index/kogito-data-index-postgresql.feature]: I just wanted some harmonization there. I think both have the `Datasource '<default>': The connection attempt failed`

Comment (radtriste) [reply tests/features/data-index/kogito-data-index-postgresql.feature]:
```suggestion
    And container log should contain Datasource '<default>': The connection attempt failed
    And container log should not contain Application failed to start
```

File: tests/features/jobs-service/kogito-jobs-service-postgresql.feature [hunk 5]
```diff
@@ -23,5 +23,5 @@ Feature: Kogito-jobs-service-postgresql feature.
       | QUARKUS_DATASOURCE_USERNAME  | kogito                                   |
       | QUARKUS_DATASOURCE_PASSWORD  | s3cr3t                                   |
     Then container log should contain -Dquarkus.http.host=0.0.0.0 -Dquarkus.http.port=8080 -jar /home/kogito/bin/jobs-service-postgresql-runner.jar
-    And container log should contain The connection attempt failed
-    And container log should contain Failed to start application (with profile prod)
+    And container log should contain org.postgresql.util.PSQLException: The connection attempt failed
+    And container log should not contain Application failed to start
```
Comment (radtriste) [inline tests/features/jobs-service/kogito-jobs-service-postgresql.feature]:
```suggestion
    And container log should contain Datasource '<default>': The connection attempt failed
    And container log should contain Failed to start application (with profile prod)
```

**Comment by radtriste** [comment]
jenkins retest this
